File: js/example/print.rb
$:.unshift(File.dirname(__FILE__) + '/../../spec')
require 'gherkin/js_lexer'

class SimplePrettyPrinterListener
  def comment(value, line)
    puts(value)
  end
  def tag(value, line)
    puts(value)
  end
  def feature(keyword, name, description, line)
    puts(keyword + ': ' + name)
    puts('  ' + description) if description.to_s != ""
  end
  def background(keyword, name, description, line)
    puts('  ' + keyword + ': ' + name)
    puts('  ' + description) if description.to_s != ""
  end
  def scenario(keyword, name, description, line)
    puts('  ' + keyword + ': ' + name)
    puts('  ' + description) if description.to_s != ""
  end
  def scenario_outline(keyword, name, description, line)
    puts('  ' + keyword + ': ' + name)
    puts('  ' + description) if description.to_s != ""
  end
  def examples(keyword, name, description, line)
    puts('  ' + keyword + ': ' + name)
    puts('    ' + description) if description.to_s != ""
  end
  def step(keyword, name, line)
    puts('    ' + keyword + name)
  end
  def py_string(string, line)
    puts('      """\n' + string + '\n      """')
  end
  def row(row, line)
    puts('      | ' + row.join(' | ') + ' |')
  end
  def eof
    puts('=====')
  end
end

Gherkin::JsLexer['en'].new(SimplePrettyPrinterListener.new).scan(IO.read(ARGV[0]))

File: js/example/print.js
var fs = require('fs');
var Lexer = require('../lib/gherkin').Lexer('en');

var lexer = new Lexer({
  comment: function(value, line) {
    console.log(value);
  },
  tag: function(value, line) {
    console.log(value);
  },
  feature: function(keyword, name, description, line) {
    console.log(keyword + ': ' + name);
  },
  background: function(keyword, name, description, line) {
    console.log('  ' + keyword + ': ' + name);
  },
  scenario: function(keyword, name, description, line) {
    console.log('  ' + keyword + ': ' + name);
  },
  scenario_outline: function(keyword, name, description, line) {
    console.log('  ' + keyword + ': ' + name);
  },
  examples: function(keyword, name, description, line) {
    console.log('  ' + keyword + ': ' + name);
  },
  step: function(keyword, name, line) {
    console.log('    ' + keyword + name);
  },
  doc_string: function(string, line) {
    console.log('      """\n' + string + '\n      """');
  },
  row: function(row, line) {
    console.log('      | ' + row.join(' | ') + ' |');
  },
  eof: function() {
    console.log('=====');
  }
});

lexer.scan(fs.readFileSync(process.ARGV[2]));
